src/os2datascanner/engine/datascanner/queue_runner.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+
+import argparse
+
+from .model.zip import ZipSource
+from .model.core import Source, SourceManager, UnknownSchemeError
+from .model.file import FilesystemSource
+from .processors import processors
+
+from queue import Full, Empty
+from ctypes import c_ulong
+from datetime import datetime
+from traceback import print_exc
+from multiprocessing import Value, Manager, Process, current_process
+
+queue = []
+
+done = "__SENTINEL__FINISHED__"
```

Comment:
The canonical way to use a sentinel is this:
```suggestion
done = object()
```
A pure instance of `object` is only equal and identical to itself.

Reply:
I don't think this works here, does it? Sentinel values are passed from process to process through queues, and that means serialising and unserialising them, and thus creating different objects.

Sentinel detection certainly doesn't work after this change:
```
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/alec/branches/magenta-aps/os2datascanner/src/os2datascanner/engine2/queue_runner.py", line 211, in <module>
    main()
  File "/home/alec/branches/magenta-aps/os2datascanner/src/os2datascanner/engine2/queue_runner.py", line 175, in main
    explore(start, sm, sources, handles)
  File "/home/alec/branches/magenta-aps/os2datascanner/src/os2datascanner/engine2/queue_runner.py", line 74, in explore
    for handle in source.handles(sm):
AttributeError: 'object' object has no attribute 'handles'
```

Reply:
> I don't think this works here, does it? Sentinel values are passed from process to process through queues, and that means serialising and unserialising them, and thus creating different objects.
> 
> Sentinel detection certainly doesn't work after this change:
> [snip]

Oh, I see — I didn't dig terribly deep into the code, just noticed that you had a variable called “sentinel”, and that's normally an `object()`.